apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -0,0 +1,513 @@
+                    $"[{operation.Name}], AverageLatency(us), {operation.Latencies.DefaultIfEmpty(0).Average() * 1000}");
+                Console.WriteLine($"[{operation.Name}], LatencyVariance(us), {Variance(operation.Latencies) * 1000}");
+                Console.WriteLine(
+                    $"[{operation.Name}], MinLatency(us), {operation.Latencies.DefaultIfEmpty(0).Min() * 1000}");
+                Console.WriteLine(
+                    $"[{operation.Name}], MaxLatency(us), {operation.Latencies.DefaultIfEmpty(0).Max() * 1000}");
+                Console.WriteLine(
+                    $"[{operation.Name}], 50thPercentileLatency(us), {sortedlatencies.Skip((int) Math.Floor(operation.Latencies.Count * .5)).First() * 1000}");
+                Console.WriteLine(
+                    $"[{operation.Name}], 95thPercentileLatency(us), {sortedlatencies.Skip((int) Math.Floor(operation.Latencies.Count * .05)).First() * 1000}");
+                Console.WriteLine(
+                    $"[{operation.Name}], 99thPercentileLatency(us), {sortedlatencies.Skip((int) Math.Floor(operation.Latencies.Count * .01)).First() * 1000}");
+                Console.WriteLine(
+                    $"[{operation.Name}], 99.9thPercentileLatency(us), {sortedlatencies.Skip((int) Math.Floor(operation.Latencies.Count * .001)).First() * 1000}");
+                Console.WriteLine($"[{operation.Name}], Return=OK, {operation.Latencies.Count}");
+                //bucket.
+                var sortedLatencies = operation.Latencies.OrderBy(x => x).ToList();
+                var latencyIndex = 0;
+                var bucket = 0;
+                for (; bucket < GetOption<long>("histogram.buckets"); bucket++)
+                {
+                    var count = 0;
+                    while (latencyIndex < sortedLatencies.Count && sortedLatencies[latencyIndex] <= bucket)
+                    {
+                        count++;
+                        latencyIndex++;
+                    }
+                    Console.WriteLine($"[{operation.Name}], {bucket}, {count}");
+                }
+                Console.WriteLine($"[{operation.Name}], >{bucket}, {sortedLatencies.Count - latencyIndex}");
+            }
+            Logger.DefaultLogger.LogPerformanceData();
+        }
+
+        internal static string GenerateRandomString()
+        {
+            s_rand.Value.NextBytes(s_byteBuffer.Value);
+            return Convert.ToBase64String(s_byteBuffer.Value);
+        }
+
+        private static void InitializeSettings()
+        {
+            AppDomain.CurrentDomain.UnhandledException += (sender, eventArgs) =>
+            {
+                Console.Error.WriteLine($"ERROR {eventArgs.ExceptionObject.ToString()}");
+            };
+
+            Trace.Listeners.Add(new TextWriterTraceListener(Console.Out));
+            if (s_debugMode)
+            {
+                Logger.DefaultLogger.LogPerformanceTraces = true;
+                if (GetOption<int>("debugmode") == 2)
```

Comment:
Any reason "debugmode" isn't extracted as a string constant like other options?

Reply:
likely because I defined it versus it being defined across all languages for ycsb.  I'll add it as a const.